src/main/scala/com/github/zenpie/macrowave/internal/Grammar.scala

```diff
@@ -57,6 +58,30 @@ final class Grammar(val c: whitebox.Context) {
   private[internal] val followSet = mutable.Map[parser.Symbol, mutable.Set[parser.TerminalSymbol]]()
   private[internal] val nullable  = mutable.Set[parser.NonTerminalSymbol]()
 
+  def dumpSymbolSets(): Unit = {
+    for ((_, id) <- namedNonTerminals) {
+      val name = nonTerminalNames(id)
+      val symbol = NonTerminalSymbol(id)
+
+      val followString = followSet(symbol).map {
+        case TokenSymbol(id) => terminalNames(id)
+        case x => x.toString
+      }.mkString("{ ", ", ", " }")
+
+      val firstString = firstSet(symbol).map {
```

Comment:
The implementation of `followString` and `firstString` are almost the same. Could you please  generalized these implementations with a local function?

Reply:
I'll do that for you, because I want to resolve this PR.